src/clib/pio_darray.c

```diff
@@ -2555,20 +2555,12 @@ if (file->iotype != PIO_IOTYPE_HDF5)
 }
 
 #ifdef _ADIOS2
-#define MAX_LEN_MATCH 64 /* Match first 64 elements only so far */
-static bool match_decomp_part(const int64_t *decomp, size_t offset, PIO_Offset *start, PIO_Offset len_decomp)
+static inline bool match_decomp_full(const int64_t *decomp, size_t offset, PIO_Offset *start, PIO_Offset len_decomp)
 {
-    PIO_Offset len = len_decomp;
-    if (len > MAX_LEN_MATCH)
-        len = MAX_LEN_MATCH;
+    assert(decomp != NULL && start != NULL);
+    assert(sizeof(int64_t) == sizeof(PIO_Offset));
```

Comment:
Is there a way to avoid this by passing in PIO_Offset *decomp (Using PIO_Offset * instead of int64_t in this case in the calling_code/rest_of_the_code)?

Reply:
What is the performance impact when you use memcmp() for entire decomps for the ne1024 case (decomps that are long)?

Reply:
> Is there a way to avoid this by passing in PIO_Offset *decomp (Using PIO_Offset * instead of int64_t in this case in the calling_code/rest_of_the_code)?

We can add a FIXME comment to handle it later.

Reply:
> What is the performance impact when you use memcmp() for entire decomps for the ne1024 case (decomps that are long)?

We will rerun some ne1024 cases on Frontier to confirm that the read performance is not significantly impacted by this change.